src/compute/src/render/join/mz_join_core.rs

```diff
@@ -234,7 +276,24 @@ where
                         for batch1 in input1_buffer.drain(..) {
                             // Ignore any pre-loaded data.
                             if PartialOrder::less_equal(&acknowledged1, batch1.lower()) {
-                                if !batch1.is_empty() {
+                                trace!(
+                                    operator_id,
+                                    input = 1,
+                                    lower = ?batch1.lower().elements(),
+                                    upper = ?batch1.upper().elements(),
+                                    "loading batch",
+                                );
+
+                                if batch1.is_empty() {
+                                    trace!(operator_id, input = 1, "skipping empty batch");
+                                } else {
+                                    trace!(
+                                        operator_id,
+                                        input = 1,
+                                        acknowledged2 = ?acknowledged2.elements(),
+                                        "deferring work for batch",
+                                    );
```

Comment:
Alternatively, log the batch len in the first `trace` call. That way you might need less trace calls.

Reply:
Done